Content.Shared/Atmos/Components/GasMinerComponent.cs

```diff
@@ -0,0 +1,60 @@
+using Robust.Shared.Serialization;
+using Robust.Shared.GameStates;
+
+namespace Content.Shared.Atmos.Components;
+
+[NetworkedComponent]
+[AutoGenerateComponentState]
+[RegisterComponent]
+public sealed partial class GasMinerComponent : Component
+{
+    /// <summary>
+    ///     Operational state of the miner.
+    /// </summary>
+    [AutoNetworkedField]
+    [ViewVariables(VVAccess.ReadOnly)]
+    public GasMinerState MinerState = GasMinerState.Disabled;
+
+    /// <summary>
+    ///      If the number of moles in the external environment exceeds this number, no gas will be mined.
+    /// </summary>
+    [ViewVariables(VVAccess.ReadWrite)]
+    [DataField]
+    public float MaxExternalAmount = float.PositiveInfinity;
+
+    /// <summary>
+    ///      If the pressure (in kPA) of the external environment exceeds this number, no gas will be mined.
+    /// </summary>
+    [ViewVariables(VVAccess.ReadWrite)]
+    [DataField]
+    public float MaxExternalPressure = Atmospherics.GasMinerDefaultMaxExternalPressure;
+
+    /// <summary>
+    ///     Gas to spawn.
+    /// </summary>
+    [ViewVariables(VVAccess.ReadWrite)]
+    [DataField]
+    public Gas? SpawnGas = null;
+
+    /// <summary>
+    ///     Temperature in Kelvin.
+    /// </summary>
+    [ViewVariables(VVAccess.ReadWrite)]
+    [DataField]
+    public float SpawnTemperature = Atmospherics.T20C;
+
+    /// <summary>
+    ///     Number of moles created per second when the miner is working.
+    /// </summary>
+    [ViewVariables(VVAccess.ReadWrite)]
+    [DataField]
+    public float SpawnAmount = Atmospherics.MolesCellStandard * 20f;
+}
+
+[Serializable, NetSerializable]
+public enum GasMinerState : byte
+{
+    Disabled,
+    Idle,
+    Working,
+}
```

Comment:
I would rename `Disabled` to something else to make it more clear what it is, or at least put an xmldoc summary on it. Before this, enabled/disabled worked more like an on/off switch, the system just returned early if the miner was disabled. Now it seems to mean that the tile is an invalid environment, and it reenables itself if the tile becomes valid again.

Reply:
Before this the `Disabled` boolean was actually kindof unused - it was implemented but it could only be accessed through VV (I guess as an admin prank or something). I'm not worried about changing the meaning of it because of that. The external behavior of an object (i.e. from the players perspective) is completely unchanged.